1 The Physical/Guess Point Level 1 (Stateless)/2 - Arrange-Act-Assert Backwards/2.1 - Password Validator/src/index.spec.ts

```diff
@@ -1,11 +1,16 @@
 type PasswordValidation = { status: boolean, errors: Array<string> };
 
+function hasUppercase(password: string) {
+    const uppercasePatternRegex: RegExp = /.*[A-Z].*/g;
+    return password.match(uppercasePatternRegex) === null;
+}
+
 function passwordValidator(password: string): PasswordValidation {
-    const uppercasePatternRegex = /.*[A-Z].*/g;
-    if (password.match(uppercasePatternRegex) === null) {
+
+    if (hasUppercase(password)) {

```

Comment:
Excellent job so far, the only thing I would mention here is that this should be named ```doesNotHaveUppercase```, right ? It actually checks if there is no uppercase.

Reply:
:+1: good point thank you